[yuechen2001] iP

yuechen2001 opened the pull request:
# Jimmy 
> "You only think of the best comeback when you leave." - Jimmy Fallon

Jimmy frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from here (to be continued).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yuechen2001 committed:
Level 0. Rename, Greet, Exit

yuechen2001 committed:
Level 1. Echo

yuechen2001 committed:
Level 2. Add, List

yuechen2001 committed:
Level 3. Mark as Done

yuechen2001 committed:
Level 4. ToDos, Events, Deadlines

yuechen2001 committed:
Level 4. ToDos, Events, Deadlines

yuechen2001 committed:
Automated Text UI Testing

yuechen2001 committed:
Level 5. Handle Errors

yuechen2001 committed:
Level 6. Delete

yuechen2001 committed:
Level 7. Save

yuechen2001 committed:
Level 8. Dates and Times

yuechen2001 committed:
Merge branch 'branch-Level-8'

yuechen2001 committed:
A-MoreOOP: Use More OOP

yuechen2001 committed:
A-Packages: Organise into Packages

yuechen2001 committed:
Merge branch 'branch-A-packages'

yuechen2001 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yuechen2001 committed:
A-Gradle: Use Gradle

yuechen2001 committed:
Add checkstyle files

yuechen2001 committed:
A-JUnit: Add JUnit Tests

yuechen2001 committed:
A-Jar: Create a JAR File

yuechen2001 committed:
A-CodingStandard: Follow the Coding Standard

yuechen2001 committed:
Level 9. Find

peienlim commented on src/main/java/essentials/TaskList.java:
```diff
@@ -0,0 +1,256 @@
+     * @param taskIndex The index of the task to be marked as incomplete.
+     * @throws JimmyException If the task index is invalid.
+     */
+    public void markTaskIncomplete(String taskIndex) throws JimmyException {
```
Perhaps this could be named as unmarkTask() instead of markTaskIncomplete task? And the variable incompleteTask could be named as taskToUnmark instead which may be more intuitive!

yuechen2001 replied:
Thank you for your suggestion, I have renamed the variables to make them clearer!

peienlim commented on src/main/java/essentials/Storage.java:
I like how the methods and variables are named in this class, very clear cut and easy to understand what they are being used for!

peienlim commented on src/main/java/essentials/TaskList.java:
```diff
@@ -0,0 +1,256 @@
+     * @throws JimmyException If the task index is invalid.
+     */
+    public void deleteTask(String taskIndex) throws JimmyException {
+        int deleteTask;
```
Perhaps it may be better to rename the deleteTask variable to something else, such as taskToDelete, since deleteTask is already the name of the method.

yuechen2001 replied:
Thank you for your suggestion, I have renamed the variables to make them clearer!

peienlim commented on src/main/java/essentials/TaskList.java:
```diff
@@ -0,0 +1,256 @@
+        taskList.add(newTask);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(newTask);
+        System.out.println(generateListCounter() + "\n");
```
I see that you have a UI.java file, so maybe you could have a function in that file which takes the task as an argument and prints the lines required when createNewTask is called?

yuechen2001 replied:
I have shifted the printing into the Ui class, thanks for the heads up.

peienlim commented on src/main/java/Jimmy.java:
```diff
@@ -0,0 +1,82 @@
+                    String instruction = parsedInput[0];
+                    String details = parsedInput[1];
+                    switch (instruction) {
+                    case "bye":
```
I see that your indentation when using switch is correct and follows the coding style guide. I find that this is a common mistake that a lot of people make, so its good that you avoided this mistake!

peienlim reviewed (commented):
Overall I can see that you have put in considerable amount of effort in the iP! Coding standards are well-followed and as a result I find your code to be straightforward and easy to understand. Just a few nits to fix, but other than that LGTM! Good job :)

baskargopinath commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/Jimmy.java` file, right-click it, and choose `Run Jimmy.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
Could this be summarized further?

yuechen2001 replied:
Yes I am planning to rewrite the README.md with GFMD this week!

baskargopinath commented on src/main/java/essentials/TaskList.java:
```diff
@@ -0,0 +1,256 @@
+
+    /**
+     * Creates a new deadline task.
+     *
```
I like the method names and variable names which are very good and easy to understand

baskargopinath commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,51 @@
+Here are the tasks in your list:
+1.[T][X] eat breakfast
+2.[T][X] go for lunch
+
```
Very nice formatting of the list with line breaks!

baskargopinath commented on src/test/java/essentials/TaskListTest.java:
```diff
@@ -0,0 +1,34 @@
+    @Test
+    public void testCreateNewTask_invalidInstruction_exceptionThrown() {
+        String invalidTaskType = "invalid";
+        String details = "invalid task details";
```
I like the setup you use to test task creation

baskargopinath commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,91 @@
+     * @param end   End time of the event.
+     * @return true if the start and end timing are valid, false otherwise.
+     */
+    private boolean checkValidDuration(LocalDate start, LocalDate end) {
```
Very appropriate and functional method names

baskargopinath reviewed (commented):
LGTM! The quality of code and attention to detail is commendable

yuechen2001 committed:
Improve code based on PR Reviews

yuechen2001 committed:
Update README.md

yuechen2001 committed:
Level 10. GUI

yuechen2001 committed:
Update README.md

yuechen2001 committed:
A-Assertions: Use Assertions

Code did not contain assertions previously.

Adding assertions to the code help to verify the programme's runtime state and assumptions.

Assertions have been added liberally as per requirements. Added assertions are mostly used to ensure that method parameters are not null.

yuechen2001 committed:
A-CodeQuality: Improve Code Quality A-CodeQuality

Code quality is not up to standard based on SE guide.

Code quality needs to improve for better readability and flow.

I have refactored parts of the code following the conventions suggested by the SE guide used by the module. Some examples include rewriting header comments and changing variable / method names.

yuechen2001 committed:
Merge pull request #2 from yuechen2001/branch-A-Assertions

A-Assertions: Use Assertions

yuechen2001 committed:
A-CI: Set up CI

yuechen2001 committed:
Merge branch 'master' into branch-A-CI

yuechen2001 committed:
Merge branch 'master' into branch-A-CodeQuality

yuechen2001 committed:
Merge pull request #3 from yuechen2001/branch-A-CodeQuality

A-CodeQuality: Improve Code Quality A-CodeQuality

yuechen2001 committed:
Merge branch 'branch-A-CI'

yuechen2001 committed:
B-ViewSchedules: View Schedules

yuechen2001 committed:
Remove IO redirection tests in CI

yuechen2001 committed:
Merge pull request #5 from yuechen2001/branch-A-CI

Remove IO redirection tests in CI

yuechen2001 committed:
Merge pull request #4 from yuechen2001/branch-B-ViewSchedules

B-ViewSchedules: View Schedules

yuechen2001 committed:
Fix variables declaration order

yuechen2001 committed:
Remove IO redirection tests again

yuechen2001 committed:
Merge pull request #6 from yuechen2001/branch-B-ViewSchedules

Fix variables declaration order

yuechen2001 committed:
Update README.md

yuechen2001 committed:
Update docs folder to build website

yuechen2001 committed:
Add title of application using JavaFX

yuechen2001 committed:
Create _config.yml

yuechen2001 committed:
Add Ui.png to website

yuechen2001 committed:
Shift _config.yml to root of directory

yuechen2001 committed:
Minor bug fixes

yuechen2001 committed:
Fix typo in error message
